### PR TITLE
Metrics Server: Cloud Run Infrastructure

### PR DESCRIPTION
**Pull request by pdewilde**

### CODE REVIEW

File: terraform/cloudrun.tf [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      dynamic "env" {
+        for_each = merge(local.default_run_envvars, var.metrics_envvars)
+
+        content {
+          name  = env.key
+          value = env.value
+        }
+      }
+    }
+    # TODO: Do I need to explicitly specify service_account or is default behavior acceptable?
```
Comment (gjonathanhong) [inline terraform/cloudrun.tf:44]: Short answer, yes best practice is to create a new service account.

Default service account can have `Editor` permissions, if this policy (https://github.com/abcxyz/infra-gcp/blob/main/terraform/org/policies/main.tf#L18C6-L18C53) is not in place.

File: terraform/cloudrun.tf [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  }
+
+  depends_on = [
+    google_project_service.services["run.googleapis.com"],
```
Comment (gjonathanhong) [inline terraform/cloudrun.tf:48]: The `run.googleapis.com` isn't included in the specified resource

File: terraform/cloudrun.tf (hunk 1 above)
Comment (pdewilde) [reply terraform/cloudrun.tf]: Added

File: terraform/cloudrun.tf [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+locals {
+  default_run_envvars = {}
+}
+
+resource "google_cloud_run_v2_service" "metrics" {
+  project = var.project_id
+
+  name         = var.metrics_service_name
+  location     = var.metrics_service_location
+  ingress      = "INGRESS_TRAFFIC_ALL"
```
Comment (gjonathanhong) [inline terraform/cloudrun.tf:25]: When you try to apply this, you will likely run into a policy [violation](https://github.com/abcxyz/infra-gcp/blob/main/terraform/org/policies/main.tf#L226) that prevents you from doing this. 

You will need to either...
- create a load balancer
- with a justification, create a policy exemption to allow all ingress

File: terraform/lb.tf [hunk 4]
```diff
@@ -136,3 +147,40 @@ resource "google_compute_backend_bucket" "default" {
     google_project_service.services["compute.googleapis.com"],
   ]
 }
+
+resource "google_compute_region_network_endpoint_group" "metrics_neg" {
+  project = var.project_id
+
+  region                = var.compute_region
+  name                  = "grc-neg"
```
Comment (gjonathanhong) [inline terraform/lb.tf:155]: 🤔

**Review by bradegler — APPROVED** [review]

File: terraform/lb.tf (hunk 4 above)
Comment (pdewilde) [reply terraform/lb.tf:155]: Doh!

**Pull request merged**